ACM#9987: Adding relase notes entries for hosted control planes

xenolinux opened the pull request:
Adding whats new entries for:

1. https://issues.redhat.com/browse/ACM-9987: Managing node pools on hosted clusters 
2. https://issues.redhat.com/browse/ACM-9621: Configuring HCP on non bare metal agent machines
3. https://issues.redhat.com/browse/ACM-9850: Creating a hosted cluster with the KubeVirt platform by using the console
4. https://issues.redhat.com/browse/ACM-9255: Adding description for the Cluster Os Image field

Additional information: The content is approved by SMEs in the main PRs:
1. https://github.com/stolostron/rhacm-docs/pull/6122
2. https://github.com/stolostron/rhacm-docs/pull/6073
3. https://github.com/stolostron/rhacm-docs/pull/6133
4. https://github.com/stolostron/rhacm-docs/pull/6098

xenolinux committed:
ACM#9987: Adding relase notes entries for hosted control planes

swopebe commented on clusters/release_notes/whats_new.adoc:
```diff
@@ -8,14 +8,26 @@ Learn more about what is new this release:
 [#hosted-control-plane]
 == Hosted control planes
+
+* Technology Preview: You can provision a hosted control plane cluster by using the non bare metal agent machines. For more information, see xref:../hosted_control_planes/non_bm_intro.adoc#configuring-hosting-service-cluster-configure-agent-non-bm[Configuring hosted control plane clusters using non bare metal agent machines].
```
wonder if there is a better way to write or say non bare metal?

xenolinux replied:
It's tricky. Agent provisioning flow is today limited to Bare metal machines. The goal is to remove the bare metal restriction, allowing any machine to be provisioned using the agent flow. To stress on the fact that "any machine can be provisioned using the Agent" the possible general term "non bare metal agent machines" is used. But in future, if I see the opportunity to use the better term, I will enhance the documentation.

swopebe committed:
Update whats_new.adoc

just bolding the tp label

swopebe reviewed (approved):
looks great, one comment I wanted to chat just your opinion, but otherwise perfect.

xenolinux commented:
Thank you for the review. I addressed all the suggestions and comments -- I will proceed with merging this PR. This content is intended to go in the OpenShift docs 4.15 welcome page (OCP 4.15 docs merge freeze date: 22nd Feb 2024)

If there are any minor comments/suggestions from the peer review perspective, I will make sure to open a follow up PR to address those changes.
